Utils/Scanner.swift

```diff
@@ -0,0 +1,175 @@
+                metadataOutput.setMetadataObjectsDelegate(viewController, queue: DispatchQueue.main)
+                metadataOutput.metadataObjectTypes = metaObjectTypes()
+            }
+            
+        } catch {
+            return nil
+        }
+        
+        return captureSession
+    }
+    
+    private func createPreviewLayer(with captureSession: AVCaptureSession, view: UIView) -> AVCaptureVideoPreviewLayer {
+        let previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = .resizeAspectFill
+        
+        return previewLayer
+    }
+    
+    // Set here the code types you want to scan
+    private func metaObjectTypes() -> [AVMetadataObject.ObjectType] {
+        return [.aztec,
+                .code128,
+                .code39,
+                .code39Mod43,
+                .code93,
+                .dataMatrix,
+                .ean13,
+                .ean8,
+                .face,
+                .interleaved2of5,
+                .itf14,
+                .pdf417,
+                .qr,
+                .upce]
+    }
+    
+    // MARK: - Public methods
+    
+    func startCaptureSession() {
+        if let captureSession = _captureSession, !captureSession.isRunning {
+            captureSession.startRunning()
+        }
+    }
+    
+    func stopCaptureSession() {
+        if let captureSession = _captureSession, captureSession.isRunning {
+            captureSession.stopRunning()
+        }
+    }
+    
+    // MARK: - Delegate methods
+    
+    func scannerDelegate(_ output: AVCaptureMetadataOutput,
+                         didOutput metadataObjects: [AVMetadataObject],
+                         from connection: AVCaptureConnection) {
+        stopCaptureSession()
+        
+        guard !metadataObjects.isEmpty, let readableObject = metadataObjects.first as? AVMetadataMachineReadableCodeObject else {
+            return
+        }
+        
+        guard let stringValue = readableObject.stringValue else { return }
+        
+        _codeOutputHandler(stringValue)
+    }
+}
+
+class ScannerViewController: UIViewController, AVCaptureMetadataOutputObjectsDelegate {
+    
+    // MARK: Properties
+    
+    private var _scanner: Scanner?
+    
+    // MARK: - Initializers
+    
+    init() {
+        _scanner = Scanner(withViewController: self,
+                           view: view,
+                           codeOutputHandler: handleCode)
+    }
+    
+    // MARK: - Camera Session handler
+    
+    // Call this method when you need to start the camera
+    func startScanner() {
+        _scanner?.startCaptureSession()
+    }
+    
+    // Call this method when you need to stop the camera
+    func stopScanner() {
+        _scanner?.stopCaptureSession()
+    }
+    
+    private func handleCode(code: String) {
+        // Add the logic that needs to be executed when any code is scanned
```

Comment:
you could add a property `onCodeScanned: ((String) -> Void)? ` that gets executed here